pkg/collector/corechecks/embed/jmx/state.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build jmx
+
+package jmx
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+type jmxState struct {
+	configs     *cache.BasicCache
+	runnerError chan struct{}
+	runner      *runner
+	lock        *sync.Mutex
+}
+
+var state jmxState = jmxState{
+	configs:     cache.NewBasicCache(),
+	runnerError: make(chan struct{}),
+	runner:      &runner{},
+	lock:        &sync.Mutex{},
+}
+
+func (s *jmxState) scheduleCheck(c *JMXCheck) error {
+	s.lock.Lock()
```

Comment:
`defer s.lock.Unlock()` would work well with this logic. Also, we could end up with a deadlock situation if return `err`. It's generally regarded as best practice unless you need to unlock in the function.